# **Problem Statement 1:**
Blood glucose levels for obese patients have a mean of 100 with a standard deviation of
15.A researcher thinks that a diet high in raw cornstarch will have a positive effect on
blood glucose levels. A sample of 36 patients who have tried the raw cornstarch diet
have a mean glucose level of 108. Test the hypothesis that the raw cornstarch had an
effect or not.

In [9]:
import scipy.stats as sts
from scipy.stats import norm
import math
import numpy as np

In [10]:
pop_mean = 100
pop_std = 15
n = 36
sample_mean = 108
alpha = 0.05

SE = pop_std/n**0.5
print(f"SE: {SE}")
Z_score = (sample_mean-pop_mean)/SE
print(f"Z_score: {Z_score}")

SE: 2.5
Z_score: 3.2


- By looking at z- table and p-value associated with 3.20 is 0.9993 
- The probability of having value less than 108 is 0.9993 and more than or equals to 108 is (1-0.9993)=0.0007.

- Step-4: Sice the probability of having mean glucose level more than or equals to 108  is 0.0007 which is  less than 0.05 
- So we will reject the Null hypothesis i.e. there is raw cornstarch effect.

# **Problem Statement 2:**
In one state, 52% of the voters are Republicans, and 48% are Democrats. In a second
state, 47% of the voters are Republicans, and 53% are Democrats. Suppose a simple
random sample of 100 voters are surveyed from each state.
What is the probability that the survey will show a greater percentage of Republican
voters in the second state than in the first state?

In [11]:
n1 = 100             # Sample Size of State 1
n2 = 100             # Sample Size of State 2
R1 = 0.52            # Percentage of Republicans from state 1
D1 = 0.48            # Percentage of Democrats from state 1
R2 = 0.47            # Percentage of Republicans from state 2
D2 = 0.53            # Percentage of Democrats from state 2

mu = R1 - R2
print(f"mu: {mu}")
std = math.sqrt(((R1 * D1 ) / n1) + ((R2 * D2) /n2))
print(f"std: {std}")
x = 0
Z_R1_R2 = ( x - mu)/std
print(f"Z_p1_p2 : {Z_R1_R2}")



mu: 0.050000000000000044
std: 0.07061869440877536
Z_p1_p2 : -0.7080278164104213


###This problem requires us to find the probability that p1 is less than p2. 
###This is equivalent to finding the probability that R1 - R2 < 0. 
###To find this probability, we need to transform the random variable (R1 - R2) into a z-score. 
###That transformation appears below.

- From Z table we find that the probability of a z-score being -0.7082 or less is 0.24.
- Therefore, the probability that the survey will show a greater percentage of Republican voters 
- In the second state than in the first state is 0.24

# **Problem Statement 3:**
You take the SAT and score 1100. The mean score for the SAT is 1026 and the standard
deviation is 209. How well did you score on the test compared to the average test taker?

In [12]:
x = 1100 #
mu = 1026 # Population Mean
sd = 209 #population standard deviation
z = ( x - mu)/sd
print("Z Score : ",z)
#the above calculation shows that my score is 0.35 standard deviations above the mean
print(f"My Score is in the range {mu - sd} - {mu + sd}  with a  zscore {z}")

Z Score :  0.35406698564593303
My Score is in the range 817 - 1235  with a  zscore 0.35406698564593303


- The z score tells you how many standard deviations from the mean your score is